crates/rustc_codegen_spirv/src/linker/structurizer.rs

```diff
@@ -177,8 +158,24 @@ fn find_block_index_from_id(blocks: &[Block], id: &Word) -> usize {
     panic!("Failed to find block from id {}", id);
 }
 
+macro_rules! get_block_mut {
+    ($builder:expr, $idx:expr) => {
+        &mut get_blocks_mut($builder)[$idx]
+    };
+}
+
+macro_rules! get_block_ref {
+    ($builder:expr, $idx:expr) => {
+        &get_blocks_ref($builder)[$idx]
+    };
+}
+
+fn idx_to_id(builder: &mut Builder, idx: usize) -> Word {
+    get_blocks_ref(builder)[idx].label_id().unwrap()
```

Comment:
gosh I'm so happy I added `label_id()`, going through that huge mess of a chain is so annoying